File: 프로그래머스/lv2/42746. 가장 큰 수/가장 큰 수.py
def solution(numbers):
    answer = ''
    s_nums = []
    for i in numbers:
        s_nums.append(str(i))
    
    s_nums.sort(key = lambda x : x*3, reverse = True)
    
    answer = str(int("".join(s_nums)))
        
    return answer

File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.java
import java.util.*;

class Solution {
    public String solution(int[] numbers) {
        String answer = "";
        
        String[] nums = new String[numbers.length];
        
        for(int i=0; i<nums.length; i++){
            nums[i] = String.valueOf(numbers[i]);
        }
            
        Arrays.sort(nums, (x, y) -> {
            return (y+x).compareTo(x+y);
        });
            
        StringBuilder sb = new StringBuilder();
        for(String n : nums)
            sb.append(n);
        
        answer = String.valueOf(sb);
        
        if (nums[0].equals("0")) {
            return "0";
        }
        
        return answer;
    }
}